pkg/agent/controller/networkpolicy/fqdn.go

```diff
@@ -405,80 +410,100 @@ func (f *fqdnController) deleteRuleSelectedPods(ruleID string) error {
 
 func (f *fqdnController) onDNSResponse(
 	fqdn string,
-	responseIPs map[string]net.IP,
-	lowestTTL uint32,
+	newDNSresponseIPs map[string]ipWithTTL,
 	waitCh chan error,
 ) {
-	if len(responseIPs) == 0 {
+	if len(newDNSresponseIPs) == 0 {
 		klog.V(4).InfoS("FQDN was not resolved to any addresses, skip updating DNS cache", "fqdn", fqdn)
 		if waitCh != nil {
 			waitCh <- nil
 		}
 		return
 	}
-	// mustCacheResponse is only true if the FQDN is already tracked by this
-	// controller, or it matches at least one fqdnSelectorItem from the policy rules.
-	// addressUpdate is only true if there has been an update in IP addresses
-	// corresponded with the FQDN.
-	mustCacheResponse, addressUpdate := false, false
-	recordTTL := time.Now().Add(time.Duration(lowestTTL) * time.Second)
+
+	currentTime := time.Now()
+	ipWithTTLMap := make(map[string]ipWithTTL)
+	minTimeToReQuery := time.Unix(1<<63-62135596801, 999999999)
```

Comment:
what time is this supposed to be? At the very least there should be a comment to explain. But based on usage, I am not sure why we are not just using the zero value for `time.Time`:
```go
var minTimeToReQuery time.Time
```

Reply:
@antoninbas Actually , 
```go
minTimeToReQuery := time.Unix(1<<63-62135596801, 999999999)
```
i had chosen above expression to establish a reference point for tracking the minimum time to re-query DNS requests as IPs expire. Since our goal in this PR for this bug/issue is to send requests as and when an IP expires, initializing ```minTimeToReQuery``` to a maximum time value allows us to  compare and update it as we process various time values , ultimately allowing us to determine the earliest re-query time.

Reply:
What you could do though is to make `earlierOf` take time.Time pointers as parameters, and return the non-nil value if one of t1/t2 is nil. That way you would not need to initialize `minTimeToReQuery`.  Note that in that case the return value should also be a pointer IMO, and when it's used eventually you should check that `minTimeToReQuery` is not nil

Reply:
Thanks @Dyanngg , yes i think using that approach we can avoid initialising ```minTimeToReQuery``` .

Reply:
I am fine with the pointer solution, but it seems to me that we could just pick a reasonable time such as `now + 24*time.Hour` and use that (with an appropriate comment of course). The time is only used if `len(ipWithTTLMap) > 0` anyway. And it seems reasonable to resend a query after 24h even if we don't really need to? At least it won't impact correctness.

Reply:
Yes, i think this was the point i was missing . Maybe i had set it to too large a value under assumption that i needed a _very _large__ value. The 24 hour time frame seems more reasonable as a replacement to existing large value that i had put.